guardian/transcription-service
# Add transcription to worker

paired with: @philmcmahon 
## What does this change?

This PR does the following:
- Gets file from S3 based on the S3 key that was sent through by the SQS message
- Creates a docker container based on the docker image `ghcr.io/guardian/transcription-service` 
- Converts the file to wav using ffmpeg within the container
- Gets the duration of the file from ffmpeg result
- Updates the `VisibilityTimeout` of the SQS message based on the duration of the file
- Transcribes a file using whisper.cpp within the container (sets the number of threads to use during computation)
- Adds the transcription result to the message that is published to SNS topic
- Set message visibility timeout to 0 if the worker fails to process the job

**TODO in upcoming PR:**
- The scaling policy rule is not working properly, so when a message is added to the queue, the ASG is not automatically triggered. This will be fixed in an upcoming PR
- The worker needs to add scale-in protection to the instance when starting to process 
- The worker needs to remove scale-in protection when it finished the work or when it fails


## How to test


Tested locally and in PROD
- Send a message through api endpoint 
- increase the desired instance of the ASG from 0 to 1
- wait to receive the notification or ssh into the instance and check the logs by `journalctl -u transcription-service-worker`
